tests/test-lcfs.c

```diff
@@ -0,0 +1,81 @@
+#define _GNU_SOURCE
+
+#include "lcfs-writer.h"
+#include <assert.h>
+#include <errno.h>
+
+static inline void lcfs_node_unrefp(struct lcfs_node_s **nodep)
+{
+	if (*nodep != NULL) {
+		lcfs_node_unref(*nodep);
+		*nodep = NULL;
+	}
+}
+#define cleanup_node __attribute__((cleanup(lcfs_node_unrefp)))
+
+static ssize_t write_cb(void *_file, void *buf, size_t count)
+{
+	FILE *file = _file;
+
+	return fwrite(buf, 1, count, file);
+}
+
+static int testwrite_node(struct lcfs_node_s *node)
+{
+	char *bufp = NULL;
+	size_t bufsz = 0;
+	FILE *buf = open_memstream(&bufp, &bufsz);
+
+	struct lcfs_write_options_s options = { 0 };
+	options.format = LCFS_FORMAT_EROFS;
+	options.version = 1;
+	options.max_version = 1;
+	options.file = buf;
+	options.file_write_cb = write_cb;
+
+	int r = lcfs_write_to(node, &options);
+	int saved_errno = errno;
```

Comment:
It took me a real long time to convince myself that this was right, but indeed we set `errno = EINVAL` in the mode-validation code.  Is there some story behind the various approaches to error handling in libcomposefs?  The mount code seems to prefer returning negative errnos, at least internally...

Reply:
> The mount code seems to prefer returning negative errnos, at least internally...

It's quite possible that I or someone else messed up the sign at some point :cry: